Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,58 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            //check for self-contradiction
+            for(int i=0; i<exactPostTime.Length-1; i++)

```

Comment:
Multiple recommendations here:
1. I would recommend simplifying the implementation since the inner for loop seems unnecessary.
2. `showPostTime[i].Equals(showPostTime[j]) == false` is a more readable and simpler way of writing `!(showPostTime[i].Equals(showPostTime[j])`
3. This is a good area for moving to a separate method and well as putting a comment to indicate what is your intent. This is because you don't want readers of the code to have to mentally process it to understand what it is trying to do. 
Example
``` csharp
private bool IsValidInput(string[] exactPostTime, string[] showPostTime)
{
    // We want to ensure that the input does not contain conflicting mappings of exactTime vs showTime. 
    // We also know that the times are synced by position. 
    // Approach
    // 1. Iterate across the times and add a mapping of exactPostTime[i] => showPostTime[i].
    // 2. As we add new mappings, if we encounter an existing mapping with conflicting values then we know that the input is invalid.

    var map = new Dictionary<string, string>();
    for(var i=0; i<exactPostTime.Length; i++)
    {
        var exists = map.TryGetValue(exactPostTime[i], out string mapping);
        if( exists == false )
            map[ exactPostTime[i] ] = showPostTime[i];
        else 
        {
            if( showPostTime[i].Equals(mapping) == false )
                return false;
        }
    }
    return true;
}
```

Reply:
Sir is it possible in single loop without using dictionary method TryGetValue? Since method itself will increase complexity

Reply:
Sir is it possible in single loop without using dictionary method TryGetValue? Since method itself will increase complexity